static/app/components/events/interfaces/spans/spanTree.tsx

```diff
@@ -141,252 +129,99 @@ class SpanTree extends React.Component<PropType> {
-      span.start_timestamp - previousSiblingEndTimestamp >= 0.1 &&
-      shouldIncludeGap;
-
-    const spanGroupNumber = isValidGap && isSpanDisplayed ? spanNumber + 1 : spanNumber;
-
     type AccType = {
-      renderedSpanChildren: Array<JSX.Element>;
-      nextSpanNumber: number;
       numOfSpansOutOfViewAbove: number;
       numOfFilteredSpansAbove: number;
-      previousSiblingEndTimestamp: undefined | number;
+      spanTree: React.ReactNode[];
     };
 
-    const treeDepthEntry = isOrphanSpan(span)
-      ? ({type: 'orphan', depth: treeDepth} as OrphanTreeDepth)
-      : treeDepth;
-
-    const treeArr = isLast
-      ? continuingTreeDepths
-      : [...continuingTreeDepths, treeDepthEntry];
-
-    const reduced: AccType = spanChildren.reduce(
-      (acc: AccType, spanChild, index) => {
-        const key = `${getSpanTraceID(span)}${getSpanID(spanChild)}`;
-
-        const results = this.renderSpan({
-          spanNumber: acc.nextSpanNumber,
-          isLast: index + 1 === spanChildren.length,
-          continuingTreeDepths: treeArr,
-          treeDepth: treeDepth + 1,
-          numOfSpansOutOfViewAbove: acc.numOfSpansOutOfViewAbove,
-          numOfFilteredSpansAbove: acc.numOfFilteredSpansAbove,
-          span: spanChild,
-          childSpans,
-          generateBounds,
-          previousSiblingEndTimestamp: acc.previousSiblingEndTimestamp,
-        });
-
-        acc.renderedSpanChildren.push(
-          <React.Fragment key={key}>{results.spanTree}</React.Fragment>
-        );
-
-        acc.numOfSpansOutOfViewAbove = results.numOfSpansOutOfViewAbove;
-        acc.numOfFilteredSpansAbove = results.numOfFilteredSpansAbove;
-
-        acc.nextSpanNumber = results.nextSpanNumber;
-
-        acc.previousSiblingEndTimestamp = spanChild.timestamp;
-
-        return acc;
-      },
-      {
-        renderedSpanChildren: [],
-        nextSpanNumber: spanGroupNumber + 1,
-        numOfSpansOutOfViewAbove: isCurrentSpanHidden ? numOfSpansOutOfViewAbove + 1 : 0,
-        numOfFilteredSpansAbove: isCurrentSpanFilteredOut
-          ? numOfFilteredSpansAbove + 1
-          : isCurrentSpanHidden
-          ? numOfFilteredSpansAbove
-          : 0,
-        previousSiblingEndTimestamp: undefined,
-      }
-    );
-
-    const infoMessage = this.generateInfoMessage({
-      isCurrentSpanHidden,
-      numOfSpansOutOfViewAbove,
-      isCurrentSpanFilteredOut,
-      numOfFilteredSpansAbove,
-    });
+    const {spanTree, numOfSpansOutOfViewAbove, numOfFilteredSpansAbove} = spans.reduce(
+      (acc: AccType, payload: EnhancedProcessedSpanType, index) => {
+        const {span, type} = payload;
+
+        switch (payload.type) {
+          case 'filtered_out': {
+            acc.numOfFilteredSpansAbove += 1;
+            return acc;
+          }
+          case 'out_of_view': {
+            acc.numOfSpansOutOfViewAbove += 1;
+            return acc;
+          }
+          default: {
+            // continue
```

Comment:
nit:

```suggestion
            break;
```

Or even if statements might be cleaner here